code/game/area/areas.dm

```diff
@@ -98,11 +98,20 @@
 /area/proc/is_no_crew_expected()
 	return flags & NO_CREW_EXPECTED
 
-/area/proc/set_lightswitch(var/state)
-	lightswitch = state
+/area/proc/set_lightswitch(var/state) // Set lights in area. TRUE for on, FALSE for off, NULL for initial state.
 	var/obj/machinery/light_switch/L = locate() in src
+	var/new_state = null
+	if(isnull(state))
+		if(L)
+			new_state = L.initial_state
```

Comment:
The initial state of an `/obj/machinery/light_switch` is defined by the area, so we only need to check the area, not the lightswitches within to find the initial state of this area.

https://github.com/Aurorastation/Aurora.3/blob/0e283455de4d83b2be826ae6c7310ec6294793c2/code/game/machinery/lightswitch.dm#L27

Unless I'm missing something here?

Reply:
Yeah, I missed that part. Should be fixed. At least the change is smaller and nicer.